pkg/tsdb/influxdb/response_parser.go

```diff
@@ -114,36 +131,52 @@ func (rp *ResponseParser) buildSeriesNameFromQuery(row Row, column string) strin
 	return fmt.Sprintf("%s.%s%s", row.Name, column, tagText)
 }
 
-func (rp *ResponseParser) parseTimepoint(valuePair []interface{}, valuePosition int) (plugins.DataTimePoint, error) {
-	value := rp.parseValue(valuePair[valuePosition])
-
-	timestampNumber, ok := valuePair[0].(json.Number)
+func parseTimestamp(value interface{}) (time.Time, error) {
+	timestampNumber, ok := value.(json.Number)
 	if !ok {
-		return plugins.DataTimePoint{}, fmt.Errorf("valuePair[0] has invalid type: %#v", valuePair[0])
+		return time.Time{}, fmt.Errorf("timestamp-value has invalid type: %#v", value)
 	}
-	timestamp, err := timestampNumber.Float64()
+	timestampFloat, err := timestampNumber.Float64()
 	if err != nil {
-		return plugins.DataTimePoint{}, err
+		return time.Time{}, err
 	}
 
-	return plugins.DataTimePoint{value, null.FloatFrom(timestamp * 1000)}, nil
+	// currently in the code the influxdb-timestamps are requested with
+	// seconds-precision, meaning these values are seconds
+	t := time.Unix(int64(timestampFloat), 0).UTC()
+
+	return t, nil
 }
 
-func (rp *ResponseParser) parseValue(value interface{}) null.Float {
+func parseValue(value interface{}) (*float64, error) {
+	// NOTE: we use pointers-to-float64 because we need
+	// to represent null-json-values. they come for example
+	// when we do a group-by with fill(null)
+
+	// FIXME: the value of an influxdb-query can be:
+	// - string
+	// - float
+	// - integer
+	// - boolean
+	//
+	// here we only handle numeric values. this is probably
+	// enough for alerting, but later if we want to support
+	// arbitrary queries, we will have to improve the code
+
+	if value == nil {
+		// this is what json-nulls become
+		return nil, nil
+	}
+
 	number, ok := value.(json.Number)
 	if !ok {
-		return null.FloatFromPtr(nil)
+		return nil, fmt.Errorf("value has invalid type: %#v", value)
 	}
 
 	fvalue, err := number.Float64()
 	if err == nil {
-		return null.FloatFrom(fvalue)
-	}
-
-	ivalue, err := number.Int64()
-	if err == nil {
-		return null.FloatFrom(float64(ivalue))
+		return &fvalue, nil
```

Comment:
I'd go with:

```
if err != nil {
		return nil, err
}

return &fvalue, nil
```

more common pattern to check if error is nil if not necessarily needed to do otherwise.